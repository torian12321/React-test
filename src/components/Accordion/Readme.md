#### Basic
```
<Accordion>Content</Accordion>
```

#### Custom title
```
<Accordion title='Custom title'>Content</Accordion>
```

#### Open by default
```
<Accordion isOpen>Content</Accordion>
```
