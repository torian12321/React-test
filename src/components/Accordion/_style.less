@import '~LESS_conf';

@color_main: @themeColor_2;
@color_sec : @themeColor_1;
@icon_w    : 14px;
@icon_h    : 3px;

.accordion{
	margin    : 10px 0;
	width     : 100%;
	overflow  : hidden;
	background: @color_main;
    font-size : @fontSize_base;
	.rounded();
	.transition();
	.smart-text-color(@color_main);
	.acc-header{		
		@gutter_x : 10px;

        font-size    : 1.2em;
        line-height  : 2;
		position     : relative;
		width        : 100%;
		padding      : 0 @gutter_x 0 ((@gutter_x * 2) + @icon_w);
		cursor       : pointer;
		background   : @color_main, 10%;
		border-bottom: @color_main, 10% 2px solid;
		.transition(background border-color);
		.smart-text-color(@color_main);
		&:before,
        &:after{
			content   : '';
            position  : absolute;
			top       :  ~"calc(1em - ((@{icon_h}) / 2))";
			left      : @gutter_x;
			width     : @icon_w;
			height    : @icon_h;
            .smart-bg-color(@color_main);
            .rounded(@icon_h);
			.transition(transform);
		}
        &:before{.rotate(90deg);}
        &:after {.rotate(180deg);}
	}
	.acc-body{
		position  : relative;
		max-height: 0;
		.transition(max-height);
		& > .acc-body-content{
			padding: 20px 10px;
		}
	}

	&:hover{
		.shadow();
	}
	&.active{
		background: lighten(@color_sec, 5%);
		.smart-text-color(lighten(@color_sec, 5%));
		.acc-header{
			background  : @color_sec;
			border-color: darken(@color_sec, 5%);
            &:before,
            &:after{
                .rotate(0);
            }
		}
		.acc-body{
			max-height: 500px;
		}
	}
}
