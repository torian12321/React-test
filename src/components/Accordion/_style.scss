@import '../../styles/conf/index.scss';

// Vars
$accordion-font-size : $fontSize !default;
$accordion-color-main: $themeColor_2 !default;
$accordion-color-sec : $themeColor_1 !default;
$accordion-icon-w    : 14px !default;
$accordion-icon-h    : 3px !default;

// Block
.accordion{
  margin    : 10px 0;
  width     : 100%;
  overflow  : hidden;
  background: $accordion-color-main;
  font-size : $accordion-font-size;
  @include rounded();
  @include transition();
  @include smart-text-color($accordion-color-main);

  // Elements
	&__header {		
		$gutter_x : 10px;

    font-size  : 1.2em;
    line-height: 2;
    position   : relative;
    width      : 100%;
    padding    : 0 $gutter_x 0 (($gutter_x * 2) + $accordion-icon-w);
    cursor     : pointer;

    // @include transition(background border-color);
    @include smart-text-color($accordion-color-main);
    &:before,
    &:after {
      content : '';
      position: absolute;
      top     : calc(1em - (#{$accordion-icon-h}) / 2);
      left    : $gutter_x;
      width   : $accordion-icon-w;
      height  : $accordion-icon-h;

      @include rounded();
      @include transition(transform);
      @include smart-bg-color($accordion-color-main);
		}
    &:before{@include rotate(90deg);}
    &:after {@include rotate(180deg);}
	}
	&__body {
		position  : relative;
		max-height: 0;
		@include transition(max-height);
    
    &__content {
			padding: 20px 10px;
		}
	}

  // Modifiers
  &:hover {
    @include shadow();
  }
  &--active {
    background: lighten($accordion-color-sec, 5%);
    @include smart-text-color(lighten($accordion-color-sec, 5%));

    .accordion {
      &__header {
        background  : $accordion-color-sec;
        border-color: darken($accordion-color-sec, 5%);

        &:before,
        &:after {
          @include rotate(0);
        }
      }
      &__body {
        max-height: 500px;
      }
    }
  }
}