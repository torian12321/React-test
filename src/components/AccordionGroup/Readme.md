#### Accordion Group
Container for multiple <Accordion /> components.
```
<AccordionGroup className='accordion-group'>
  <Accordion title="Accordion 1">Content</Accordion>
  <Accordion title="Accordion 2">Content</Accordion>
  <Accordion title="Accordion 3" isOpen>Content</Accordion>
  <Accordion title="Accordion 4">Content</Accordion>
</AccordionGroup>
```
