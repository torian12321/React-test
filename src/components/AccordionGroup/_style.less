@import '~LESS_conf';

.accordion-group{
	margin: 0 10px;
	.accordion{
		margin: 0;
		border-bottom: 2px solid @themeColor_1;
		.rounded(0);
		&:first-child{
			.rounded(@BRadious; @BRadious; 0; 0;);
		}
		&:last-child{
			.rounded(0; 0; @BRadious; @BRadious;);
			border-bottom: 0;
		}
	}
}