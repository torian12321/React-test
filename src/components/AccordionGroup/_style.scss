@import '../../styles/conf/index.scss';

// Vars
$accordionGroup-border-r: $border-radious !default;

// Block
.accordion-group{
  margin: 0 10px;

  .accordion{
    margin       : 0;
    border-bottom: 2px solid saturate($themeColor_2, 30);
    @include rounded(0);

    &--active{
      border-color: $themeColor_1;
    }
    
    &:first-child{
      @include rounded($accordionGroup-border-r $accordionGroup-border-r 0 0);
    }
    &:last-child{
      @include rounded(0 0 $accordionGroup-border-r $accordionGroup-border-r);
      border-bottom: 0;
    }
  }
}