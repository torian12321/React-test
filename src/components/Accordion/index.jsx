import React      from 'react';
import PropTypes  from 'prop-types';
import classNames from 'classnames';
import './_style';

class Accordion extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpen    : this.props.isOpen,
      body_style: {}
    };
  }
  componentWillReceiveProps(nextProps) {
    if (nextProps.isOpen !== this.state.isOpen) {
      this.setState({isOpen: nextProps.isOpen});
    }
  }
    
  expand(){
    const accContent_h = this.accContent.clientHeight;

    this.setState({
      isOpen    : !this.state.isOpen,
      body_style: {
        maxHeight: !this.state.isOpen ? accContent_h : 0
      }
    });
  }

  render() {
    return (
      <div className={classNames('accordion', {'active': this.state.isOpen})}>
        <div className="acc-header" onClick= {this.expand.bind(this)}>
          {this.props.title}
        </div>
        <div className="acc-body" style={this.state.body_style}>
          <div className="acc-body-content" ref={(c) => { this.accContent = c; }}>
            {this.props.children}
          </div>
        </div>
      </div>
    );
  }
}

Accordion.propTypes = {
  /** Accordion content. */
  children: PropTypes.any.isRequired,
  title   : PropTypes.string,
  isOpen  : PropTypes.bool
};
Accordion.defaultProps = {
  title : 'Show Details',
  isOpen: false
};

module.exports = Accordion;
