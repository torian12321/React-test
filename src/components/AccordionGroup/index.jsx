import React      from 'react';
import PropTypes  from 'prop-types';
import classNames from 'classnames';
import './_style.less';

class AccordionGroup extends React.Component {
  render() {
    return (
      <div className={classNames('accordion-group', this.props.className)}>
        {this.props.children}
      </div>
    )
  }
}

AccordionGroup.propTypes = {
  children : PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.element
  ]).isRequired, 
  className: PropTypes.string
};

module.exports = AccordionGroup;
