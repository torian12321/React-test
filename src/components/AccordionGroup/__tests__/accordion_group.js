import React          from 'react';
import renderer       from 'react-test-renderer';
import Accordion      from '../../Accordion';
import AccordionGroup from '../index';

describe('component', () => {
  describe('<AccordionGroup />', () => {

    it('should render correctly', () => {
      expect(renderer.create(
        <AccordionGroup className='accordion-group'>
          <Accordion title="Accordion 1">Content 1</Accordion>
          <Accordion title="Accordion 2" isOpen>Content 2</Accordion>
          <Accordion title="Accordion 3" isOpen>Content 3</Accordion>
        </AccordionGroup>
      )).toMatchSnapshot();

    });
  });
});
