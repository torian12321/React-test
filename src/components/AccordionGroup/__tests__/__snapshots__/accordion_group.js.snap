// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <AccordionGroup /> should render correctly 1`] = `
<div
  className="accordion-group accordion-group"
>
  <div
    className="accordion"
  >
    <div
      className="acc-header"
      onClick={[Function]}
    >
      Accordion 1
    </div>
    <div
      className="acc-body"
      style={Object {}}
    >
      <div
        className="acc-body-content"
      >
        Content 1
      </div>
    </div>
  </div>
  <div
    className="accordion active"
  >
    <div
      className="acc-header"
      onClick={[Function]}
    >
      Accordion 2
    </div>
    <div
      className="acc-body"
      style={Object {}}
    >
      <div
        className="acc-body-content"
      >
        Content 2
      </div>
    </div>
  </div>
  <div
    className="accordion active"
  >
    <div
      className="acc-header"
      onClick={[Function]}
    >
      Accordion 3
    </div>
    <div
      className="acc-body"
      style={Object {}}
    >
      <div
        className="acc-body-content"
      >
        Content 3
      </div>
    </div>
  </div>
</div>
`;
